Organize Windows Cairo implementation files

donny-dont opened the pull request:
#### d5cdf88dfec9c3bf12861b6a5e806f0e1d6187b8
<pre>
Organize Windows Cairo implementation files
<a href="https://bugs.webkit.org/show_bug.cgi?id=269593">https://bugs.webkit.org/show_bug.cgi?id=269593</a>

Reviewed by Brent Fulgham.

Put `PLATFORM(WIN) &amp;&amp; USE(CAIRO)` files into a `win/cairo` directory to make
it easier to determine where Cairo specific implementations live. Merge and
remove files that had no Cairo specific implementations but were named as if
they did.

* Source/WebCore/PlatformWin.cmake:
* Source/WebCore/page/win/FrameCairoWin.cpp: Removed.
* Source/WebCore/page/win/FrameWin.cpp:
* Source/WebCore/platform/graphics/win/FontCacheWin.cpp:
* Source/WebCore/platform/graphics/win/FontCustomPlatformDataWin.cpp:
* Source/WebCore/platform/graphics/win/FontWin.cpp:
* Source/WebCore/platform/graphics/win/GlyphPageTreeNodeWin.cpp: Renamed from Source/WebCore/platform/graphics/win/GlyphPageTreeNodeCairoWin.cpp.
* Source/WebCore/platform/graphics/win/GraphicsContextWin.cpp:
* Source/WebCore/platform/graphics/win/ImageAdapterWin.cpp:
* Source/WebCore/platform/graphics/win/MediaPlayerPrivateMediaFoundation.cpp:
* Source/WebCore/platform/graphics/win/SimpleFontDataCairoWin.cpp: Removed.
* Source/WebCore/platform/graphics/win/SimpleFontDataWin.cpp:
* Source/WebCore/platform/graphics/win/cairo/FontCacheWinCairo.cpp: Copied from Source/WebCore/platform/graphics/win/SimpleFontDataWin.cpp.
* Source/WebCore/platform/graphics/win/cairo/FontCustomPlatformDataWinCairo.cpp: Added.
* Source/WebCore/platform/graphics/win/cairo/FontPlatformDataWinCairo.cpp: Renamed from Source/WebCore/platform/graphics/win/FontPlatformDataCairoWin.cpp.
* Source/WebCore/platform/graphics/win/cairo/GraphicsContextWinCairo.cpp: Renamed from Source/WebCore/platform/graphics/win/GraphicsContextCairoWin.cpp.
* Source/WebCore/platform/graphics/win/cairo/ImageAdapterWinCairo.cpp: Copied from Source/WebCore/platform/graphics/win/ImageAdapterWin.cpp.
* Source/WebCore/platform/graphics/win/cairo/MediaPlayerPrivateMediaFoundationCairo.cpp: Added.
* Source/WebCore/platform/win/cairo/DragImageWinCairo.cpp: Renamed from Source/WebCore/platform/win/DragImageCairoWin.cpp.

Canonical link: <a href="https://commits.webkit.org/274920@main">https://commits.webkit.org/274920@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/b6e7c8daaa6bf7bf2c8f2a954d427fa6ef7f3342

| Misc | iOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/40331 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/48/builds/19343 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/14/builds/42709 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/42876 "Built successfully") | [✅ 🛠 wincairo](https://ews-build.webkit.org/#/builders/32/builds/36420 "Built successfully") 
| [✅ 🧪 bindings](https://ews-build.webkit.org/#/builders/9/builds/42638 "Passed tests") | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/49/builds/22279 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/51/builds/16672 "Built successfully") | [✅ 🧪 wpe-wk2](https://ews-build.webkit.org/#/builders/34/builds/33469 "Passed tests") | 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/40905 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/47/builds/16288 "Passed tests") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/18/builds/34780 "Passed tests") | [✅ 🧪 api-wpe](https://ews-build.webkit.org/#/builders/41/builds/14047 "Passed tests") | 
| [✅ 🧪 webkitpy](https://ews-build.webkit.org/#/builders/6/builds/40200 "Passed tests") | [✅ 🧪 ios-wk2-wpt](https://ews-build.webkit.org/#/builders/42/builds/14099 "Passed tests") | [✅ 🧪 mac-wk1](https://ews-build.webkit.org/#/builders/10/builds/35735 "Passed tests") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/44154 "Built successfully") | 
| [✅ 🛠 🧪 jsc](https://ews-build.webkit.org/#/builders/20/builds/33784 "Built successfully and passed tests") | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/13/builds/36555 "Passed tests") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/36/builds/36063 "Passed tests") | [✅ 🧪 gtk-wk2](https://ews-build.webkit.org/#/builders/1/builds/39810 "Passed tests") | 
| [✅ 🛠 🧪 jsc-arm64](https://ews-build.webkit.org/#/builders/12/builds/39957 "Built successfully and passed tests") | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/44/builds/15146 "Built successfully") | [✅ 🧪 mac-AS-debug-wk2](https://ews-build.webkit.org/#/builders/50/builds/12401 "Passed tests") | [✅ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/21/builds/38085 "Passed tests") | 
| [✅ 🧪 services](https://ews-build.webkit.org/#/builders/28/builds/40378 "Passed tests") | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/46/builds/16765 "Built successfully") | [✅ 🧪 mac-wk2-stress](https://ews-build.webkit.org/#/builders/8/builds/35018 "Passed tests") | [✅ 🛠 jsc-armv7](https://ews-build.webkit.org/#/builders/35/builds/46966 "Built successfully") | 
| [✅ 🛠 🧪 merge](https://ews-build.webkit.org/#/builders/19/builds/9061 "Built successfully and passed tests") | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/43/builds/16814 "Built successfully") | | [✅ 🧪 jsc-armv7-tests](https://ews-build.webkit.org/#/builders/25/builds/9660 "Passed tests") | 
| | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/45/builds/16409 "Built successfully") | | | 
<!--EWS-Status-Bubble-End-->

fujii reviewed (commented):
We will remove these code soon. I don't like an idea moving these files.

brentfulgham reviewed (approved):
Looks great! r=me

fujii commented on Source/WebCore/platform/graphics/win/MediaPlayerPrivateMediaFoundation.cpp:
```diff
@@ -2796,53 +2795,6 @@ HRESULT MediaPlayerPrivateMediaFoundation::Direct3DPresenter::presentSample(IMFS
     return hr;
 }
 
-void MediaPlayerPrivateMediaFoundation::Direct3DPresenter::paintCurrentFrame(WebCore::GraphicsContext& context, const WebCore::FloatRect& destRect)
```
We shouldn't move this method. Using #if USE(CAIRO) and USE(SKIA) is better.

fujii replied:
Are you planning to add MediaPlayerPrivateMediaFoundationSkia.cpp? After finish skia migration, will you remove the MediaPlayerPrivateMediaFoundationSkia.cpp?

donny-dont committed:
Organize Windows Cairo implementation files
https://bugs.webkit.org/show_bug.cgi?id=269593

Reviewed by Brent Fulgham.

Put `PLATFORM(WIN) && USE(CAIRO)` files into a `win/cairo` directory to make
it easier to determine where Cairo specific implementations live. Merge and
remove files that had no Cairo specific implementations but were named as if
they did.

* Source/WebCore/PlatformWin.cmake:
* Source/WebCore/page/win/FrameCairoWin.cpp: Removed.
* Source/WebCore/page/win/FrameWin.cpp:
* Source/WebCore/platform/graphics/win/FontCacheWin.cpp:
* Source/WebCore/platform/graphics/win/FontCustomPlatformDataWin.cpp:
* Source/WebCore/platform/graphics/win/FontWin.cpp:
* Source/WebCore/platform/graphics/win/GlyphPageTreeNodeWin.cpp: Renamed from Source/WebCore/platform/graphics/win/GlyphPageTreeNodeCairoWin.cpp.
* Source/WebCore/platform/graphics/win/GraphicsContextWin.cpp:
* Source/WebCore/platform/graphics/win/ImageAdapterWin.cpp:
* Source/WebCore/platform/graphics/win/MediaPlayerPrivateMediaFoundation.cpp:
* Source/WebCore/platform/graphics/win/SimpleFontDataCairoWin.cpp: Removed.
* Source/WebCore/platform/graphics/win/SimpleFontDataWin.cpp:
* Source/WebCore/platform/graphics/win/cairo/FontCacheWinCairo.cpp: Copied from Source/WebCore/platform/graphics/win/SimpleFontDataWin.cpp.
* Source/WebCore/platform/graphics/win/cairo/FontCustomPlatformDataWinCairo.cpp: Added.
* Source/WebCore/platform/graphics/win/cairo/FontPlatformDataWinCairo.cpp: Renamed from Source/WebCore/platform/graphics/win/FontPlatformDataCairoWin.cpp.
* Source/WebCore/platform/graphics/win/cairo/GraphicsContextWinCairo.cpp: Renamed from Source/WebCore/platform/graphics/win/GraphicsContextCairoWin.cpp.
* Source/WebCore/platform/graphics/win/cairo/ImageAdapterWinCairo.cpp: Copied from Source/WebCore/platform/graphics/win/ImageAdapterWin.cpp.
* Source/WebCore/platform/graphics/win/cairo/MediaPlayerPrivateMediaFoundationCairo.cpp: Added.
* Source/WebCore/platform/win/cairo/DragImageWinCairo.cpp: Renamed from Source/WebCore/platform/win/DragImageCairoWin.cpp.

Canonical link: https://commits.webkit.org/274920@main
